cpp/src/phonenumbers/phonenumberutil.cc

```diff
@@ -1527,12 +1525,76 @@ void PhoneNumberUtil::FormatNsnWithCarrier(const string& number,
       : metadata.intl_number_format();
   const NumberFormat* formatting_pattern =
       ChooseFormattingPatternForNumber(available_formats, number);
```

Comment:
This is just one place that calls `ChooseFormattingPatternForNumber` before then attempting formatting - what about all the others?